d2l-input-time.html

```diff
@@ -0,0 +1,392 @@
+<link rel="import" href="../polymer/polymer.html">
+<link rel="import" href="d2l-input-shared-styles.html">
+<link rel="import" href="../d2l-dropdown/d2l-dropdown.html">
+<link rel="import" href="../d2l-dropdown/d2l-dropdown-content.html">
+<link rel="import" href="../iron-a11y-keys-behavior/iron-a11y-keys-behavior.html">
+<link rel="import" href="../iron-selector/iron-selector.html">
+<link rel="import" href="../d2l-typography/d2l-typography-shared-styles.html">
+<link rel="import" href="../d2l-intl-import/d2l-intl.html">
+
+<!--
+`d2l-input-time`
+Accessible, Localized Time Picker Input Element
+
+@demo demo/d2l-input-time.html
+-->
+
+<dom-module id="d2l-input-time">
+	<template strip-whitespace>
+		<style include="d2l-input-styles">
+			:host {
+				--d2l-input-line-height: 1;
+				--d2l-input-padding: 0.7rem 1rem;
+				--d2l-input-padding-focus: calc(0.7rem - 1px) calc(1rem - 1px);
+
+				display: inline-flex;
+				flex-direction: column;
+			}
+
+			input {
+				font-family: inherit;
+			}
+
+			.d2l-dropdown-opener {
+				max-width: 6rem;
+			}
+
+			d2l-dropdown-content {
+				--d2l-dropdown-verticaloffset: 10px;
+			}
+
+			.timezone {
+				@apply --d2l-body-compact-text;
+				color: var(--d2l-color-tungsten);
+				font-style: italic;
+				padding-top: 10px;
+				max-width: 6rem;
+				white-space: nowrap;
+			}
+
+			iron-selector {
+				box-sizing: border-box;
+				display: block;
+				width: 100%;
+			}
+			.d2l-input-time-item {
+				box-sizing: border-box;
+				border-top: 1px solid transparent;
+				border-bottom: 1px solid var(--d2l-color-gypsum);
+				width: calc(100% - 2px);
+				font-size: 0.8rem;
+				padding: 0.75rem 1rem;
+				cursor: pointer;
+			}
+			.d2l-input-time-item:last-of-type {
+				border-bottom-color: transparent;
+			}
+			.d2l-input-time-item:not([disabled]):focus,
+			.d2l-input-time-item:not([disabled]):hover,
+			.d2l-input-time-item:not([disabled])[selected] {
+				background-color: var(--d2l-color-celestine-plus-2);
+				border-top-color: var(--d2l-color-celestine-plus-1);
+				border-bottom-color: var(--d2l-color-celestine-plus-1);
+				color: var(--d2l-color-celestine);
+			}
+		</style>
+
+		<d2l-dropdown no-auto-open>
+			<div class="d2l-dropdown-opener" id="opener">
```

Comment:
Normally I think the actual opener (the element the user will click or focus on) should have `d2l-dropdown-opener`.  The `d2l-dropdown-opener-behavior` has some logic that I think expects this, for instance to apply focus to the opener when the use hits [Escape] inside the dropdown content.  There's likely a couple more little things like that.  So in this case, I _think_ it should be on the `input`, but perhaps there is a reason it's not in this case?

Reply:
Looks like there's an issue with the scrollbar when you do this, and there's special functionality to handle scrolling what has been selected into view.  I think it's related to this issue: https://github.com/BrightspaceUI/time-picker/issues/9

I've added https://github.com/BrightspaceUI/inputs/issues/21 with this comment.

Reply:
Yeah, I saw the scrolling bit earlier too.  That's one of the behaviors that made me wonder if this should really be using your select implementation. (?)  I'm surprised the opener would affect scrolling though.

Reply:
Yeah, I think the select component could be a really good fit.  I've been investigating a bit more but I think I'll just throw them in issues and hopefully we can just replace it later and remove those issues altogether.